### PR TITLE
Implement self-signed means

### PR DESCRIPTION
**Pull request by reinkrul**
TODO:

- [x] Add required assurance level (low) to NutsAuthorizationCredentials after https://github.com/nuts-foundation/nuts-node/pull/2179

### CODE REVIEW

File: api/auth.go [hunk 1]
```diff
@@ -140,32 +177,53 @@ func (auth *Auth) JWTHandler(next echo.HandlerFunc) echo.HandlerFunc {
 	}
 }
 
-func createPasswordVP(customerID int, password string) auth.VerifiablePresentation {
+func createPasswordVP(customerDID string) auth.VerifiablePresentation {
+	const identifier = "t.tester@example.com"
```
Comment (woutslakhorst) [inline api/auth.go:181]: Don't we have usernames in the demo to login?

**Review by woutslakhorst — COMMENTED** [review]
Would it be to much effort to remove the EmployeeCredential for a logged in user and just have its username? It's extremely weird having a VP as session without using it.....

File: api/auth.go (hunk 1 above)
Comment (reinkrul) [reply api/auth.go:181]: No, only password.

**Comment by reinkrul** [comment]
> Would it be to much effort to remove the EmployeeCredential for a logged in user and just have its username? It's extremely weird having a VP as session without using it.....

I agree the current approach is a bit weird, I just extended it. But I'm fine with reworking it so we only have a VP when actually elevated using the Nuts node.

**Comment by reinkrul** [comment]
Need to extract user info from IRMA VP, testing.

**Comment by woutslakhorst** [comment]
> Need to extract user info from IRMA VP, testing.

@reinkrul 
Yeah, I used an empty UserInfo struct in the last two commits to get everything working again. Test it on demo.woutslakhorst.nl

**Review by woutslakhorst — APPROVED** [review]
Let's tag it afterwards and test it on our setups

**Pull request merged**